generator/src/main/java/com/scottlogic/deg/generator/generation/grouped/FieldSpecGroupValueGenerator.java

```diff
@@ -17,172 +17,177 @@
 import com.scottlogic.deg.generator.fieldspecs.relations.FieldSpecRelations;
-import com.scottlogic.deg.generator.generation.FieldPair;
 import com.scottlogic.deg.generator.generation.FieldSpecValueGenerator;
 import com.scottlogic.deg.generator.generation.databags.*;
+import com.scottlogic.deg.generator.restrictions.linear.Limit;
+import com.scottlogic.deg.generator.restrictions.linear.LinearRestrictions;
+import com.scottlogic.deg.generator.restrictions.linear.LinearRestrictionsFactory;
 import com.scottlogic.deg.generator.utils.SetUtils;
 
 import java.time.OffsetDateTime;
 import java.util.*;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static com.scottlogic.deg.generator.generation.grouped.FieldSpecGroupDateHelper.adjustBoundsOfDate;
+import static com.scottlogic.deg.common.util.FlatMappingSpliterator.flatMap;
 
 public class FieldSpecGroupValueGenerator {
 
+    private final CombinationStrategyType combinationStrategy;
     private final FieldSpecValueGenerator underlyingGenerator;
+    private final FieldSpecMerger fieldSpecMerger = new FieldSpecMerger();
 
-    public FieldSpecGroupValueGenerator(FieldSpecValueGenerator underlyingGenerator) {
+    @Inject
+    public FieldSpecGroupValueGenerator(FieldSpecValueGenerator underlyingGenerator, CombinationStrategyType combinationStrategy) {
         this.underlyingGenerator = underlyingGenerator;
+        this.combinationStrategy = combinationStrategy;
     }
 
     public Stream<DataBag> generate(FieldSpecGroup group) {
-        Field first = SetUtils.firstIteratorElement(group.fieldSpecs().keySet());
+        Field first = getFirst(group);
 
-        FieldSpecGroup groupRespectingFirstField = initialAdjustments(first, group);
-        FieldSpec firstSpec = groupRespectingFirstField.fieldSpecs().get(first);
+        if (group.fieldSpecs().size() == 1){
+            return underlyingGenerator.generate(first, group.fieldSpecs().get(first))
+                .map(val -> toDataBag(first, val));
+        }
+
+        FieldSpec firstSpec = updateFirstSpecFromRelations(first, group);
 
         Stream<DataBag> firstDataBagValues = underlyingGenerator.generate(first, firstSpec)
             .map(value -> toDataBag(first, value));
 
-        return createRemainingDataBags(firstDataBagValues, first, groupRespectingFirstField);
+        return flatMap(
+            firstDataBagValues,
+            dataBag -> generateRemainingData(first, dataBag, removeSpecFromGroup(first, group)));
     }
 
-    private static DataBag toDataBag(Field field, DataBagValue value) {
-        Map<Field, DataBagValue> map = new HashMap<>();
-        map.put(field, value);
-        return new DataBag(map);
+    private Field getFirst(FieldSpecGroup keySet) {
+        Map<Field, Integer> otherCount = keySet.relations().stream().collect(Collectors.toMap(
+            r -> r.other(),
```

Comment:
r -> r.other() could be replace with FieldSpecRelations::other - again an intellij recommendation and general good practice